docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,66 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism that the compiler performs at link time. At compile time, the compiler creates a special intermediate representation of all translation units. It then optimizes them as a single unit at link time, which uses less memory than non-LTO builds.
+
+## Using LTO in Mbed OS
+
+The Mbed OS build system implements LTO as an optional profile extension in `tools\profiles\extensions\lto.json`. Enabling LTO amends the build profile with LTO flags. LTO performs heavy memory optimizations that break debugging, so it's recommended to use it only with release profile.
+
+To enable LTO add `--profile` option with LTO file path `tools\profiles\extensions\lto.json` to the build command.
+
+<span class="notes">**Note**: For profile extensions you have to put the full path relative to the project's root folder.</span>
+
+To enable LTO with the `release` profile:
+
+```
+mbed compile -t TOOLCHAIN -m TARGET --profile release --profile mbed-os/tools/profiles/extensions/lto.json
+```
+
+Example LTO profile memory savings for [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky):
+
+|Build type|Total static RAM memory (data + BSS)|Total flash memory (text + data)|
+| --- | --- | --- |
+| GCC_ARM - release - no LTO | 12,096B | 44,628B |
+| GCC_ARM - release - LTO | 11,800B | 41,088B |
+|***saved memory***|296B|3,540B|
+| ARM - release - no LTO | 10,365B | 35,496B |
+| ARM - release - LTO | 10,153B | 31,514B |
+|***saved memory***|212B|‭3,982‬B|
+
+LTO profile build results for [mbed-cloud-client-example](https://github.com/ARMmbed/mbed-cloud-client-example)
+
+|Build type|Total Static RAM memory (data + BSS)|Total Flash memory (text + data)|
+| --- | --- | --- |
+| GCC_ARM - release - no LTO | 59,760B | 389,637B |
+| GCC_ARM - release - LTO | 59,432B | 354,167B |
+|***saved memory***| 328B | ‭35,470‬B|
+| ARM - release - no LTO | 58,099B | 353,849B |
+| ARM - release - LTO | 57,150B | 322,500B |
+|***saved memory***| 949B | ‭31,349‬B|
+
+<span class="notes">**Note**: In LTO builds, the compiler produces bytecode/bitcode instead of regular object code. It's hard to analyze this output with object code analysis tools.</span>
+
+## Limitations
+
+- LTO slows down build process.
+- It’s very hard to control memory placement when using LTO.
+- LTO performs heavy memory optimizations that break debugging.
+- In LTO builds, the compiler produce bytecode/bitcode instead of regular object code. It's hard to analyze this output with object code analysis tools.
```

Comment:
```suggestion
- In LTO builds, the compiler produces bytecode/bitcode instead of regular object code. It's hard to analyze this output with object code analysis tools.
```

Reply:
We've said this already in the main text